src/middlewared/middlewared/plugins/pool.py

```diff
@@ -4069,5 +4085,51 @@ def parse_lsof(lsof, dirs):
     return list(pids.items())
 
 
+def parse_fstat(fstat, dirs):
+
+    # fstat output is separated by newlines
+    fstat = fstat.split('\n')
+
+    # first line is the column headers
+    fstat.pop(0)
+
+    # comprehension does a few things
+    #    1. remove any empty lines
+    #    2. remove any lines that dont have
+    #        at least 5 columns. The 5th
+    #        column is the path associated
+    #        with the process.
+    #    3. remove the g_eli,g_mirror information
+    #        from the output as its not needed
+    #        and ends up shifting those lines
+    #        quite a bit since the thread names
+    #        have spaces in their names
+    #    4. remove any lines that don't have a traditional
+    #        "path" since the path component for processes
+    #        can be a stream/socket/pipe etc
+    fstat = [
+        i for i in fstat if i and len(i.split()) >= 5 and not i.split()[1].startswith(('g_eli', 'g_mirror')) and '/' in i.split()[4]
+    ]
+
+    pids = {}
+    pid = command = None
+    for line in fstat:
+        line = line.split()
```

Comment:
```suggestion
    for line in fstat:
```

Reply:
With the other change this is simplified as well.
I find fields to be a better name than line but that's not a big difference either way.

Reply:
👍